File: LocalBuySellAPI/src/main/java/com/genspark/LocalBuySellAPI/Service/AccountService.java
package com.genspark.LocalBuySellAPI.Service;


import com.genspark.LocalBuySellAPI.Entity.Account;

import javax.transaction.Transactional;
import java.util.List;
import java.util.Optional;

public interface AccountService {

    Account create(Account account);

    List<Account> getAll();

    Account getById(int accountId);

    void deleteById(int accountId);

    Account editAccount(Account edited, int accountId);

    @Transactional
    Account findAccountByEmail(String email);
}

File: frontend/src/Service/AccountService.js
import axios from 'axios';

const ACCOUNT_API_URL = "http://localhost:8080/account";

class AccountService {
    
    // GET
    getAll() {
        return axios.get(ACCOUNT_API_URL + "/all");
    }

    getById(accountId) {
        return axios.get(ACCOUNT_API_URL + "/" + accountId);
    }

    // CREATE
    create(account) {
        return axios.post(ACCOUNT_API_URL + "/create", account)
    }

    // DELETE
    deleteById(accountId) {
        return axios.delete(ACCOUNT_API_URL + "/delete/" + accountId);
    }
}

export default new AccountService();